perf/rdm_bw.c

```diff
@@ -364,20 +365,26 @@ int main(int argc, char *argv[])
 	r_buf = (char *) (((unsigned long) r_buf_original + (align_size - 1)) /
 				align_size * align_size);
 
-	ret = fi_mr_reg(dom, r_buf, MYBUFSIZE, 0, 0, 0, 0, &mr, NULL);
+	ret = fi_mr_reg(dom, r_buf, MYBUFSIZE, FI_REMOTE_WRITE, 0, 0, 0, &r_mr, NULL);
 	if (ret) {
 		FT_PRINTERR("fi_mr_reg", ret);
 		return -1;
 	}
 
 	lbuf_desc.addr = (uint64_t)r_buf;
-	lbuf_desc.key = fi_mr_key(mr);
+	lbuf_desc.key = fi_mr_key(r_mr);
 
 	rbuf_descs = (buf_desc_t *)malloc(numprocs * sizeof(buf_desc_t));
 
 	/* Distribute memory keys */
 	FT_Allgather(&lbuf_desc, sizeof(lbuf_desc), rbuf_descs);
 
+	ret = fi_mr_reg(dom, s_buf, MYBUFSIZE, FI_WRITE, 0, 0, 0, &l_mr, NULL);
```

Comment:
Technically I think it should be "FI_WRITE|FI_SEND|FI_RECV" since s_buf is used as source of RMA write, and buffer for outband and inbound message transfers.